### ISSUE TITLE
Skeleton: in some cases, cannot determine if a field being accessed in a method really belongs to the enclosing class

### ISSUE THREAD

**Issue by llorllale** [yegor256/jpeek#156]
We need more than just the name to determine if a class is using its own attribute or that of another one. We should somehow include the fully qualified name of the class that owns the field that is being operated on.

To illustrate the problem, consider two classes:
```java
public final class Foo {
    private static String NAME = "hey";

    public void test() {
      Bar.NAME = "test";
    }
}
public final class Bar {
    public static String NAME = "hey";

    public void test() {
      Bar.NAME = "test";
    }
}
```
The resulting skeleton for `Foo.test()` does not tell us whether the field `NAME` is its own attribute or that of another class:
```xml
<method abstract="false"
        ctor="false"
        desc="()V"
        name="test"
        public="true"
        static="false">
   <return>V</return>
   <args/>
   <ops>
      <op code="put_static">NAME</op>
   </ops>
</method>
```

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by 0crat** [comment]
Job [#156](https://github.com/yegor256/jpeek/issues/156) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://datum.zerocracy.com/pages/policy.html#29): +15 points just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale), total is [+495](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
The job [#156](https://github.com/yegor256/jpeek/issues/156) assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/1552061599); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by paulodamaso** [comment]
@0crat please wait, looks like there is no reviewer for this issue yet.

**Comment by 0crat** [comment]
> @0crat please wait for #257 review ([here](https://github.com/yegor256/jpeek/issues/156#issuecomment-395755131))

@paulodamaso Can't understand "please", try one of these:
  - `assign` Assign a performer to this issue
  - `boost` Set a boost for the job
  - `hello` Just say hello
  - `in` Register this issue in scope (WBS)
  - `out` Close the order and remove this job from scope
  - `pay` Pay a user some extra cash
  - `quality` Review a task
  - `resign` Resign from this issue
  - `status` Check the status of the job
  - `version` Print current version of the bot
  - `wait` Register an impediment for a job

**Comment by paulodamaso** [comment]
@0crat wait for #257 review

**Comment by 0crat** [comment]
> @0crat wait for #257 review ([here](https://github.com/yegor256/jpeek/issues/156#issuecomment-395769264))

@paulodamaso The impediment for [#156](https://github.com/yegor256/jpeek/issues/156) was registered successfully by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0pdd** [comment]
@llorllale the puzzle [#261](https://github.com/yegor256/jpeek/issues/261) is still not solved.

**Comment by paulodamaso** [comment]
@llorllale #257 merged and puzzle #261 left for improving implementation, please close.

**Comment by llorllale** [comment]
@paulodamaso thanks

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#156](https://github.com/yegor256/jpeek/issues/156) is now out of scope

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Comment by 0pdd** [comment]
@llorllale the puzzle [#358](https://github.com/yegor256/jpeek/issues/358) is still not solved; solved: [#261](https://github.com/yegor256/jpeek/issues/261).

**Comment by 0pdd** [comment]
@llorllale all 2 puzzles are solved here: [#261](https://github.com/yegor256/jpeek/issues/261), [#358](https://github.com/yegor256/jpeek/issues/358).
